### DISCUSSION TITLE
Alignment with the GeoJSON spec

### DISCUSSION THREAD

**Discussion by benjamintd** [placemark/geojson-folders#1] [Ideas]
Hi @tmcw ! 

I hope this is the right place to submit comments about your proposal. 
I really like the idea, and just wanted to give a few initial thoughts about the spec. I think the nomenclature could be closer to GeoJSON with a few tweaks: 
- using PascalCase for types (`Root`, `Folder`)
- renaming the `meta` key to `properties` to further align with the place attributes are typically found in GeoJSON. 

Using your example, it could look like: 
```
{
  "type": "Root",
  "children": [
    {
      "type": "Folder",
      "properties": {
        "name": "Test"
      },
      "children": [
         // ...features and folders
      ]
    }
    // ...features
  ]
}
```

This would make the structure "feel" more like GeoJSON, if that is something this spec is concerned with.

**Comment by tmcw** [comment]
Okay, yes!

- Using PascalCase makes a lot of sense. Good reason to do it, no reason not to do it.
- `properties`… also I think is a straightforward win, in the direction of inventing less. The GeoJSON spec is a little strict about FeatureCollections and Geometries _not_ having `properties` but I don't think that means anything for whether a totally new object can have em.

**Reply by sgillies** [reply]
@tmcw what is a Folder's properties about? Is it just an anything-goes namespace for user data? Do features in a folder acquire the properties from their folder?

The reason why GeoJSON Geometry and FeatureCollection don't have properties is to help interoperability. I was pretty convinced that users would fuzz the boundaries between FeatureCollection/Feature/Geometry in non-compatible ways if we made them all feature-like. I wonder if GeoJSON Folders is in a similar place? If I have a Folder with properties and it contains a GeoJSON Geometry, would we treat it as a Feature?

**Reply by tmcw** [reply]
>  Do features in a folder acquire the properties from their folder?

No - basically this is for folder metadata, which in the most basic case is just a name. KML is a classic example - folders have names and descriptions and IDs. In GPX, you might translate each track to a folder, and in that case the GPX track has a name, usually.

> If I have a Folder with properties and it contains a GeoJSON Geometry, would we treat it as a Feature?

I hope not! Folders should have a `children` and `properties` object and nothing else. Would want to make that distinction very clear. Would calling the folder metadata `meta` or `metadata` help with this distinction, or is this something you think could just live in a strongly-worded spec?
